Add CAMARA Validation result handling to the API readiness checklist documentation (#569)
camaraproject/ReleaseManagement: documentation/readiness/api-readiness-checklist.md

> **Note**: During development on `main`, API version fields in the YAML definitions must stay as `wip`. The release automation process replaces them with the correct version numbers and applicable extensions during snapshot creation.

### Handling validation results 

> Validation results must be handled as early as possible, and at the latest before the final rc pre-release, as follows:

Reviewer: Actually I would expect the documentation of open warnings and the planned handling at every (pre-)release, that will allow to see that the team is aware of the still open work towards a public release.

With the formulation "at the latest before the final rc pre-release" a rc pre-release without the documentation would be possible ... which we should not accept.

Author: To what I remember, that is what we discussed in the rm call, but I agree with your comment and will update the text accordingly.

Author: Updated, please rereview - not fully sure about the "(controlled at each pre-release") part. 
Better wording very welcome

> **Note**: During development on `main`, API version fields in the YAML definitions must stay as `wip`. The release automation process replaces them with the correct version numbers and applicable extensions during snapshot creation.

### Handling validation results 

> Validation results must be handled as early as possible (controlled in each pre-release) as follows:

Reviewer: ```suggestion
Validation results must be handled as early as possible during each release preparation, and no later than before the final rc pre-release:
```

- Errors MUST be fixed (errors block snapshot creation).

> - Warnings MUST be fixed, or explicitly deferred by documenting them in an issue (copying the validation summary lines) with a deferral reason per line. For example, for stable APIs, a warning that would need a breaking change while no major version update is planned is a valid deferral reason.

Reviewer: ```suggestion
- Warnings MUST be fixed or explicitly deferred by documenting them in one or more issues. The issue(s) must copy the relevant validation summary lines and include the deferral reason(s). For stable APIs, a warning that would require a breaking change while no major version update is planned is a valid deferral reason.
```

> - Hints MUST be checked and MAY be fixed (hints do not block snapshot creataion nor a release-review).

Reviewer: ```suggestion
- Hints MUST be checked and MAY be fixed. Hints do not block snapshot creation or Release PR review.
```

- API definitions are correct and complete for the target status
- Test cases cover the intended API behavior
- Documentation is adequate for the target audience

> - Any remaining warnings are documented in an issue stating an explicit deferreal reason

Reviewer: ```suggestion
- Any remaining warnings are documented in one or more issues, with the relevant validation summary lines and deferral reason(s)
```

- CHANGELOG follows the release documentation rules
- Breaking changes are documented and version updates follow SemVer rules
- All mandatory release assets for the declared API(s) status(es) are present as per the checklist

> - All remaining warnings have an acceptable deferral reason

Reviewer: ```suggestion
- All remaining warnings have been documented and have acceptable deferral reasons
```

Author: this one I slightly updated.
Herbert's update was originally "All remaining warnings have documented and acceptable deferral reasons"
